src/components/Synthetics/StatusNotification/FeesSettlementStatusNotification.tsx

```diff
@@ -0,0 +1,191 @@
+import { Trans, plural, t } from "@lingui/macro";
+import cx from "classnames";
+import { TransactionStatus, TransactionStatusType } from "components/TransactionStatus/TransactionStatus";
+import {
+  OrderStatus,
+  PendingFundingFeeSettlementData,
+  PendingOrderData,
+  getPendingOrderKey,
+  useSyntheticsEvents,
+} from "context/SyntheticsEvents";
+import { MarketInfo, MarketsInfoData, getMarketIndexName, getMarketPoolName } from "domain/synthetics/markets";
+import { isMarketOrderType } from "domain/synthetics/orders";
+import { getByKey } from "lib/objects";
+import { useEffect, useMemo, useState } from "react";
+import "./StatusNotification.scss";
+import { useToastAutoClose } from "./useToastAutoClose";
+
+type Props = {
+  toastTimestamp: number;
+  orders: PendingFundingFeeSettlementData["orders"];
+  marketsInfoData: MarketsInfoData | undefined;
+};
+
+export function FeesSettlementStatusNotification({ orders, toastTimestamp, marketsInfoData }: Props) {
+  const { orderStatuses: allOrderStatuses, setOrderStatusViewed } = useSyntheticsEvents();
+
+  const [matchedOrderStatusKeys, setMatchedOrderStatusKeys] = useState<string[]>([]);
+  const matchedOrderStatuses = useMemo(
+    () => matchedOrderStatusKeys.map((key) => allOrderStatuses[key]),
+    [allOrderStatuses, matchedOrderStatusKeys]
+  );
+
+  const [keyByOrder, orderByKey] = useMemo(() => {
+    const map1 = new Map<PendingOrderData, string>();
+    const map2 = new Map<string, PendingOrderData>();
+    orders.forEach((order) => {
+      const key = getPendingOrderKey(order);
```

Comment:
just to confirm, why not use `order.key` itself?

Reply:
i guess in this case we may not need to have these two maps

Reply:
1. that is only way to connect `order` with `orderStatus` (`PendingOrderData` doesn't contain `key`)
2. `getPendingOrderKey(order) != order.key`